### DISCUSSION TITLE
Community ownership

### DISCUSSION THREAD

**Discussion by jonaslagoni** [asyncapi/modelina#487] [Ideas]
The number of programming languages, inputs, and integrations that this library can be used in is countless, and we need a way for the community to take ownership of the library. A formal way for people to engage and help drive Modelina to its absolute limit if there even is one.

In case you want the big picture for Modelina, the [features overview](https://github.com/asyncapi/modelina#features) should give you a good idea. Take any input and generate data models in any language with an extreme level of customization, as use-cases are rarely the same.

How do you see we best achieve this?

**Comment by jonaslagoni** [comment]
# Language Champions

My suggestion is that we start with the generators, and introduce **Language Champions** 🥇

I know that we are talking about the generators, but IMO `Language Champions` sound better than `Generator Champions`. Let me know otherwise 😄 

**So what does this entail?**

- You will maintain a specific output generator. It can of course also be multiple if you are up for the challenge, but primarily, with a focus on one generator.
- You will shape the generator based on the community and your own ideas and feedback.
- It will be your responsibility to review PR's and issues affecting the generator.

**What do you get out of it?**

- You will be added as a [committer](https://www.asyncapi.com/blog/governance-motivation#vocabulary) for Modelina
- You will be added as a [TSC member for AsyncAPI](https://www.asyncapi.com/community/tsc) and help make decisions on a global scale.
- You will help shape the future of Modelina, as you are of course more than welcome to help out in any other part of the library!
- You will gain knowledge about languages and collaboration that will shape your future. Collaborating is not easy, one does not decide, but a consensus has to be achieved, explaining problems and solving those problems takes time to master
- You will gain knowledge not only about building software but also how you create documentation and testing that ensures we build Modelina for others.

## FAQ
### Can multiple people be language champions for the same language?
Absolutely.

Collaboration is key ✌️

### Who can become a Language Champion?
Anyone.

Anyone who wants to learn, contribute or have the passion to collaborate 🙂 

### I don't know any of the current generators, or TypeScript, but want to help?

We provide any help in the form needed, this can be direct mentoring, community zoom meetings, etc. We will guide you through it. What matters is wanting it.

If you have any questions feel free to [join slack](https://asyncapi.com/slack-invite) and ping me a message! 

### Which generators can I champion?
Any existing, and non-existing.

Do you have a new generator you would like to champion, just create a future request and let us know 👍 

The current generators can be found in the [feature list](https://github.com/asyncapi/modelina#features)

### Why only one generator?
My suggestion is that we stick to one generator per champion because a language has enough variants and customizations that we need to enable, that you won't be shy for stuff to do 😄 

That being said, you are more than welcome to help out in any other part of the library, it is actually highly encouraged 👍 

## Would this be the "best" approach?
This is of course just a suggestion, let me know what you think!

Good, bad, unreachable, not the time yet, need better clarification?

**Reply by derberg** [reply]
LGTM!
Do you already have a list of initial Champions?

**Reply by jonaslagoni** [reply]
@panwauu or @md-saif-husain any interest here? 🙂 

Other than that, I will add individual issues listing the "open position" and readme change 🙂

**Reply by md-saif-husain** [reply]
@jonaslagoni I am interested to be typescript champion .

**Comment by boyney123** [comment]
>I know that we are talking about the generators, but IMO Language Champions sound better than Generator Champions. Let me know otherwise 😄

Yeah, I think like Language Champions sounds great.

@jonaslagoni so to clarify will the eco-system work like our generate and templates do at the moment?

For example, we will have a range of generators as you mention, will they all belong in separate repos and somehow be pulled together in this one here?

**Reply by jonaslagoni** [reply]
@boyney123 good question! For now, no, all generators are be located here: https://github.com/asyncapi/modelina/tree/master/src/generators

That being said, in the future, I would not be surprised if that was a direction we needed to take, but let's see. Letting the problems drive the change 🙂 Cause separating the generators is gonna be difficult and confusing as well 😅

**Reply by jonaslagoni** [reply]
This means from a TypeScript champion's perspective, all of these files is what you will primarily maintain are https://github.com/asyncapi/modelina/tree/master/src/generators/typescript 

But of course, I think it is inevitable that you wont make changes to other part of the library as well 😄

**Reply by boyney123** [reply]
Yeah, perfect sounds sensible.

> Letting the problems drive the change

Love it!

Yeah, I think you're spot on with the idea to be honest.

Guess as part of the project ReadME it would be good to list who the champions are too

**Reply by jonaslagoni** [reply]
Great idea yea 👍 Once we agree on the approach the contribution docs are gonna be changed as well as project readme 🙂

**Comment by jonaslagoni** [comment]
As there were no objections I took the time to propose the initial take on the concept which can be found here: https://github.com/asyncapi/modelina/pull/638

Please leave any comments and reflect on whether it makes sense 🙂

**Comment by jonaslagoni** [comment]
The concept have not been implemented: https://github.com/asyncapi/modelina/blob/master/docs/champions.md

Locking discussion.
